similarity-search/src/main/java/org/phenotips/similarity/internal/SolrSimilarPatientsFinder.java

```diff
@@ -204,4 +214,36 @@ private long count(SolrQuery query)
         }
         return 0;
     }
+
+    private Collection<String> getPresentPhenotypeTerms(Patient patient)
+    {
+        Collection<String> termIds = new ArrayList<String>();
```

Comment:
I tested it, and specifying a term multiple times favors those terms instead of the ones appearing only once. It's boosting the more generic terms, while we actually want to emphasize the more specific ones further down the ontology.